### DISCUSSION TITLE
Event message logic

### DISCUSSION THREAD

**Discussion by apaperclip** [blakeblackshear/frigate#382] [General]
Hi Blake,

Now that I have working notifications I was testing them out.  I have an automation that simply watches the front yard and will send me a notification with an attachment, this is based on the readme. I wanted to better understand when frigate will publish messages for an event. Right now I'm seeing several messages per event which results in several notifications, often in quick succession.  Not saying its a problem, I may have some values that need adjusting.

Thanks!

**Comment by blakeblackshear** [comment]
A new message with the same id is posted each time frigate determines that it has found a "better" view of the tracked object. It is looking at size, age, and score to determine what is better. I also need to add an update when the object's zones change as well. You can use the before and after values to determine what changes should trigger additional notifications. The example in the readme uses the `tag` to update the notification in place if you are using the HA companion apps.

**Reply by apaperclip** [reply]
Ok, I will look into the trigger logic.  Couple questions for you.

Is the first event message always ```false_positive: true?``` and ```end_time: nulll```
Can a running/active event have ```false_positive: true?```
Will an event always end with a non-null end_time?

I really like the idea of ongoing updates for an event. Just trying to be friendly to push notification volume. Last I recall there are limits for the service HA is using.   I thinking of an notification automation the start of an event and the end of an event.


One good example of a lot of event messages that happened today. I had an event that was 9 seconds (based on start/end times in the last event message) and has ~20 events.

On the clip side, it was just one clip, 24 seconds long (10s was pre capture)

**Reply by blakeblackshear** [reply]
The first message should always be false_positive: true in the before and false_positive: false in the after. That change is what causes event messages to start publishing. I can see how that would happen in situations where each consecutive frame has a better score or the tracked object gets larger.

**Reply by apaperclip** [reply]
Thanks for the quick reply!

If I'm still getting a ton of notifications when I go 'live' with this past my testing I think will do something like trigger on false_positive as true in either/or the before/after (need to test and see what works best/simple logic)  and then trigger again when when the end_time is not null. That should alert me right away when something starts, and it should end some reasonable amount of time later and I would get an end with the last 'best' picture.  

How does the length of an event mesh with the max recording time? Will frigate just start recording a second clip at 301s if the event is still active?

**Reply by blakeblackshear** [reply]
It will start expiring the cache from the beginning. Once the event ends, it will assemble the clip from whatever is available in the cache. The clip isn't available until the event has ended even if it is longer than the 300 second limit.
